src/components v2/Atoms/Forms/Checkbox/Checkbox.vue

```diff
@@ -0,0 +1,208 @@
+<template>
+  <label
+    class="ph-checkbox-label"
+    :class="[{ checked: value, disabled, focused }, size]"
+  >
+    <input
+      :checked="value"
+      :disabled="disabled"
+      :name="name"
+      type="checkbox"
+      class="ph-checkbox-input"
+      @input="$emit('input', $event.target.checked)"
+      @focus="onFocus"
+      @blur="onBlur"
+    />
+    <div class="ph-checkbox-toggle">
+      <p-icon
+        name="Checkmark"
+        :type="iconType[size]"
+        class="ph-checkbox-checkmark"
+      />
+    </div>
+    {{ label }}
+    <slot />
+  </label>
+</template>
+
+<script lang="ts">
+import Vue, { PropType } from 'vue';
+import PIcon from '@/components/Icon';
+import { v4 as uuidv4 } from 'uuid';
+
+export default Vue.extend({
+  name: 'P2Checkbox',
+
+  components: {
+    PIcon,
+  },
+
+  props: {
+    value: {
+      type: Boolean as PropType<boolean>,
+      default: false,
+    },
+
+    label: {
+      type: String as PropType<string>,
+      default: '',
+    },
+
+    name: {
+      type: String as PropType<string>,
+      default: '',
+    },
+
+    disabled: {
+      type: Boolean as PropType<boolean>,
+      default: false,
+    },
+
+    size: {
+      type: String as PropType<string>,
+      default: 'md',
+    },
+  },
+
+  data() {
+    return {
+      id: uuidv4(),
+      focused: false,
+      iconType: {
+        sm: 'xxs',
+        md: 'xs',
+      },
+    };
+  },
+
+  created() {
+    const $this = this as any;
+    if (!!$this.label && $this.$slots.default) {
+      console.warn(
+        'Checkbox label and Checkbox slot(default) are both defined and will be rendered together. This is probably not intended.'
+      );
+    }
+  },
+
+  methods: {
+    onFocus() {
+      this.focused = true;
+    },
+    onBlur() {
+      this.focused = false;
+    },
+  },
+});
+</script>
+
+<style lang="scss" scoped>
+.ph-checkbox-label {
+  border: 1px dashed transparent;
+  cursor: pointer;
+  display: flex;
+  position: relative;
+  transition-property: color, border-color;
+  transition-duration: 0.3s;
+  color: var(--checkbox-default-label-color);
+
+  .ph-checkbox-input {
+    position: absolute;
+    opacity: 0;
+  }
+
+  .ph-checkbox-toggle {
+    border-radius: 4px;
```

Comment:
Do we plan to make border-radius themeable? Overriding border radius is prevelant.

Reply:
No. Checkboxes should not have border-radius themable. Most people recognise a checkbox pattern with rounded corners. Even in our AMP design system where we have border-radius:0 on buttons and form inputs, the checkboxes have rounded corners. 

Eventually I'd like to see us move away from custom SVG checkboxes to using a font-awesome for a consistent iconography across all controls, and faster build time.